tests/includes/transformer/class-test-comment.php

```diff
@@ -0,0 +1,168 @@
+					return self::dummy_response(
+						wp_json_encode(
+							array(
+								'subject' => 'acct:author@remote.example',
+								'links'   => array(
+									'self' => array( 'href' => 'https://remote.example/@author' ),
+								),
+							)
+						)
+					);
+				}
+				if ( str_starts_with( $url, 'https://example.net/' ) ) {
+					return self::dummy_response(
+						wp_json_encode(
+							array(
+								'subject' => 'https://example.net/@remote',
+								'aliases' => array(
+									'acct:remote@example.net',
+								),
+								'links'   => array(
+									'self' => array( 'href' => 'https://example.net/@remote' ),
+								),
+							)
+						)
+					);
+				}
+				return $data;
+			},
+			10,
+			3
+		);
+	}
+
+	/**
+	 * Clean up after tests.
+	 */
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$post_id, true );
+	}
+
+	/**
+	 * Test content generation with reply context.
+	 *
+	 * @covers ::to_object
+	 */
+	public function test_content_with_reply_context() {
+		// Create a parent ActivityPub comment.
+		$parent_comment_id = wp_insert_comment(
+			array(
+				'comment_post_ID'      => self::$post_id,
+				'comment_content'      => 'Parent comment',
+				'comment_type'         => 'comment',
+				'comment_author'       => 'Remote Author',
+				'comment_author_url'   => 'https://remote.example/@author',
+				'comment_author_email' => '',
+				'comment_meta'         => array(
+					'protocol' => 'activitypub',
+				),
+			)
+		);
+
+		// Create a reply comment.
+		$reply_comment_id = wp_insert_comment(
+			array(
+				'comment_post_ID'      => self::$post_id,
+				'comment_parent'       => $parent_comment_id,
+				'comment_content'      => 'Reply comment',
+				'comment_type'         => 'comment',
+				'comment_author'       => 'Local Author',
+				'comment_author_url'   => 'https://example.net/@remote',
+				'comment_author_email' => '',
+				'comment_meta'         => array(
+					'protocol' => 'activitypub',
+				),
+			)
+		);
+
+		// Create a reply comment.
+		$test_comment_id = wp_insert_comment(
+			array(
+				'comment_post_ID'      => self::$post_id,
+				'comment_parent'       => $reply_comment_id,
+				'comment_content'      => 'Reply comment',
+				'comment_type'         => 'comment',
+				'comment_author'       => 'Local Author',
+				'comment_author_url'   => 'https://example.com/@test',
+				'comment_author_email' => '',
+			)
+		);
+
+		// Transform comment to ActivityPub object.
+		$comment     = get_comment( $test_comment_id );
+		$transformer = new Comment( $comment );
+		$object      = $transformer->to_object();
+
+		// Get the content.
+		$content = $object->get_content();
+
+		// Test that reply context is added.
+		$this->assertStringContainsString( 'Reply comment', $content );
+		$this->assertStringContainsString( '<a class="u-mention mention" href="https://remote.example/@author">@author@remote.example</a>', $content );
+
```

Comment:
It might be helpful to add a test with multiple mentions that also compares the entire output to make sure we're happy with whitespace.

Reply:
it already adds multiple, but I will extend the test, to check for all of them.